freqtrade/optimize/hyperopt.py

```diff
@@ -237,27 +237,63 @@ def init_spaces(self):
             logger.debug("Hyperopt has 'protection' space")
             # Enable Protections if protection space is selected.
             self.config['enable_protections'] = True
-            self.protection_space = self.custom_hyperopt.protection_space()
+            try:
+                self.protection_space = self.custom_hyperopt.protection_space()
+            except OperationalException as e:
+                if self.config["hyperopt_ignore_unparam_space"]:
+                    logger.warning(e)
+                else:
+                    raise
 
         if HyperoptTools.has_space(self.config, 'buy'):
             logger.debug("Hyperopt has 'buy' space")
-            self.buy_space = self.custom_hyperopt.buy_indicator_space()
+            try:
+                self.buy_space = self.custom_hyperopt.buy_indicator_space()
+            except OperationalException as e:
+                if self.config["hyperopt_ignore_unparam_space"]:
+                    logger.warning(e)
+                else:
+                    raise
 
         if HyperoptTools.has_space(self.config, 'sell'):
             logger.debug("Hyperopt has 'sell' space")
-            self.sell_space = self.custom_hyperopt.sell_indicator_space()
+            try:
+                self.sell_space = self.custom_hyperopt.sell_indicator_space()
+            except OperationalException as e:
+                if self.config["hyperopt_ignore_unparam_space"]:
+                    logger.warning(e)
```

Comment:
On a 2nd thought - maybe consider moving this to `hyperopt_auto.py` (although i'm not sure how to handle return value then) (it'd be a nice to have - it'll however need some further tests / experimentation to decide if that'll really work).

the exception itself is thrown out of `_format_exception_message` - so we'd have one location to check for this and return a sane default (probably a empty list?)